src/widget.ts

```diff
@@ -19,7 +19,9 @@ export class SidecarModel extends output.OutputModel {
       _view_module: SidecarModel.view_module,
       _view_module_version: SidecarModel.view_module_version,
       title: 'Sidecar',
-      anchor: 'right'
+      anchor: 'right',
+      ref: null,
+      _widget_id: null,
     };
```

Comment:
In the model, make a new Promise property that will resolve when the view was actually created.

```suggestion
    };
    
    get created() {
      return this._viewCreated.promise;
    }
    
    public resolveCreated() {
      this._viewCreated.resolve();
    }
    
    private _viewCreated = new PromiseDelegate();  // Import this from '@lumino/coreutils'
    
    // You also need to deserialize the ref property
    static serializers = {
      ...DOMWidgetModel.serializers,
      ref: { deserialize: unpack_models as any }, // import unpack_models from '@jupyter-widgets/base'
    };
```